File: ClientApp/src/app/_models/subject.ts
import { City } from "./city";
import { Route } from "./route";

export class Subject {
    id: number;
    parent_id: number;
    name: number;
    women_kid: number;
    women_adult: number;
    women_senior: number;
    men_kid: number;
    men_adult: number;
    men_senior: number;
    sum: number;
    cities: City[];
    routes: Route[];

    constructor(data: any) {
        this.id = data.id;
        this.parent_id = data.parentId;
        this.name = data.name;
        this.women_kid = data.womenKid;
        this.women_adult = data.womenAdult;
        this.women_senior = data.womenSenior;
        this.men_kid = data.menKid;
        this.men_adult = data.menAdult;
        this.men_senior = data.menSenior;
        this.sum = data.sum;
        this.cities = (data.cities as any[]).map(data => new City(data));
        this.routes = (data.routes as any[]).map(data => new Route(data));
    }
}

File: Models/Subject.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

using GridWebApp.Models.Interfaces;

namespace GridWebApp.Models
{
    //Субъект РФ
    [Table("subjects")]
    public class Subject : IModel, IData
    {
        public Subject()
        {
            Cities = new List<City>();
            Routes = new List<Route>();
        }

        [JsonProperty("REGION_ID")]
        [Key]
        public long Id { get; set; }

        [JsonProperty("PARENT_REGION_ID")]
        public int ParentId { get; set; }

        [JsonProperty("REGION_NAME")]
        public string Name { get; set; }


        [JsonProperty("VALUE_WOMEN_KID")]
        public int WomenKid { get; set; }

        [JsonProperty("VALUE_WOMEN_ADULT")]
        public int WomenAdult { get; set; }

        [JsonProperty("VALUE_WOMEN_SENIOR")]
        public int WomenSenior { get; set; }


        [JsonProperty("VALUE_MEN_KID")]
        public int MenKid { get; set; }

        [JsonProperty("VALUE_MEN_ADULT")]
        public int MenAdult { get; set; }

        [JsonProperty("VALUE_MEN_SENIOR")]
        public int MenSenior { get; set; }


        [JsonProperty("VALUE_SUM")]
        public int Sum { get; set; }

        public List<City> Cities { get; set; }

        public List<Route> Routes { get; set; }

        public void CalcSum()
        {
            Sum = MenKid + MenAdult + MenSenior + WomenKid + WomenAdult + WomenSenior;
        }
    }
}